rotkehlchen/chain/ethereum/manager.py

```diff
@@ -528,21 +530,54 @@ def ens_lookup(
             self,
             name: str,
             call_order: Optional[Sequence[NodeName]] = None,
-    ) -> Optional[ChecksumEthAddress]:
+            blockchain: SupportedBlockchain = SupportedBlockchain.ETHEREUM,
+    ) -> Optional[EnsAddress]:
```

Comment:
For the typing I absolutely hate Union types. It's a mess to deal with. IF you have to use `cast` at the place where you call it you know that you need to make changes.

So what I would suggest doing here is to do the overload of the type depending on the blockchain argument, so that mypy can properly figure out what type is returned depending on what blockchain argument is given.

Or split it into multiple functions ... but probably doing the overload should be enough and scales bettter.

By overload I mean the @overload decorator as used for example here:

https://github.com/rotki/rotki/blob/72c94fc48d62dfb53dda067c9857aafa6bcde9db/rotkehlchen/externalapis/etherscan.py#L91-L145

This way you will also be able to remove the `cast`